src/facilities/facility-unit-attributes.repository.ts

```diff
@@ -1,193 +1,227 @@
         }
-
-        string += `OR (UPPER(fua.secondaryFuelInfo) ~* ${regex}) `;
       }
 
       string += ')';
       query.andWhere(string);
     }
 
-    if (facilityAttributesParamsDTO.programCodeInfo) {
+    if (params.controlTechnologies) {
       let string = '(';
 
       for (
         let i = 0;
-        i < facilityAttributesParamsDTO.programCodeInfo.length;
+        i < params.controlTechnologies.length;
         i++
       ) {
         const regex = Regex.commaDelimited(
-          facilityAttributesParamsDTO.programCodeInfo[i].toUpperCase(),
+          params.controlTechnologies[i].toUpperCase(),
         );
 
         if (i === 0) {
-          string += `(UPPER(fua.programCodeInfo) ~* ${regex}) `;
+          string += `(UPPER (fua.so2ControlInfo) ~* ${regex}) `;
         } else {
-          string += `OR (UPPER(fua.programCodeInfo) ~* ${regex}) `;
+          string += `OR (UPPER(fua.so2ControlInfo) ~* ${regex}) `;
         }
+
+        string += `OR (UPPER(fua.noxControlInfo) ~* ${regex}) `;
+
+        string += `OR (UPPER(fua.pmControlInfo) ~* ${regex}) `;
+
+        string += `OR (UPPER(fua.hgControlInfo) ~* ${regex}) `;
       }
 
       string += ')';
-      query.andWhere(string);
-    }
 
-    if (facilityAttributesParamsDTO.sourceCategory) {
-      query.andWhere(`UPPER(fua.sourceCategory) IN (:...sourceCategories)`, {
-        sourceCategories: facilityAttributesParamsDTO.sourceCategory.map(
-          sourceCategories => {
-            return sourceCategories.toUpperCase();
-          },
-        ),
-      });
+      query.andWhere(string);
     }
 
     query
       .orderBy('fua.facilityId')
       .addOrderBy('fua.unitId')
       .addOrderBy('fua.year');
 
+    return query;
+  }
+
+  private getColumns(isStreamed: boolean): string[] {
+    const columns = [
+      'fua.id',
+      'fua.stateCode',
+      'fua.facilityName',
+      'fua.facilityId',
+      'fua.unitId',
+      'fua.associatedStacks',
+      'fua.year',
+      'fua.programCodeInfo',
+      'fua.epaRegion',
+      'fua.nercRegion',
+      'fua.county',
+      'fua.countyCode',
+      'fua.fipsCode',
+      'fua.sourceCategory',
+      'fua.latitude',
+      'fua.longitude',
+      'fua.ownDisplay',
+      'fua.oprDisplay',
+      'fua.so2Phase',
+      'fua.noxPhase',
+      'fua.unitType',
+      'fua.primaryFuelInfo',
+      'fua.secondaryFuelInfo',
+      'fua.so2ControlInfo',
+      'fua.noxControlInfo',
+      'fua.pmControlInfo',
+      'fua.hgControlInfo',
+      'fua.commercialOperationDate',
+      'fua.operatingStatus',
+      'fua.maxHourlyHIRate',
+      'fua.generatorId',
+      'fua.arpNameplateCapacity',
+      'fua.otherNameplateCapacity',
+    ]
+
+    return columns.map(col => {
+      if (isStreamed) {
+        if (col === 'fua.ownDisplay') {
+          return `${col} AS "ownerOperatorInfo"`;
+        }
+        if (col === 'fua.generatorId') {
+          return `${col} AS "associatedGeneratorsAndNamePlateCapacity"`;
```

Comment:
The alias is associatedGeneratorsAndNameplateCapacity (the p is lowcase in plate)

return `${col} AS "associatedGeneratorsAndNameplateCapacity"`;